typescript/src/deposit-sweep.ts

```diff
@@ -420,3 +500,51 @@ export async function submitDepositSweepProof(
     vault
   )
 }
+
+/**
+ * Checks if a UTXO is owned by a provided key pair based on its previous output
+ * script.
+ * @dev The function assumes previous output script comes form the P2PKH or
+ *      P2WPKH UTXO.
+ * @param keyPair - A Signer object containing the public key and private key
+ *        pair.
+ * @param prevOutScript - A Buffer containing the previous output script of the
+ *        UTXO.
+ * @param network - The Bitcoin network configuration, i.e. mainnet or testnet.
+ * @returns A boolean indicating whether the derived address from the UTXO's
+ *          previous output script matches either of the P2PKH or P2WPKH
+ *          addresses derived from the provided key pair.
+ */
+export function ownsUtxo(
+  keyPair: Signer,
+  prevOutScript: Buffer,
+  network: networks.Network
```

Comment:
Let's use `BitcoinNetwork` and do the conversion here.

Reply:
Doen in be1b1fe44aa1784fb28c630a62dcc1b10a85e0a6.